### PR TITLE
Fix volume update for recurringJob during upgrade

### PR DESCRIPTION
**Pull request by c3y1huang**
https://github.com/longhorn/longhorn/issues/2961

### CODE REVIEW

File: upgrade/v110to120/upgrade.go [hunk 1]
```diff
@@ -265,17 +266,19 @@ func translateVolumeRecurringJobsToLabel(namespace string, lhClient *lhclientset
...
 			key := types.GetRecurringJobLabelKey(types.LonghornLabelRecurringJob, id)
 			volumeLabels[key] = types.LonghornLabelValueEnabled
+			updateVolume = true
 		}
-		if len(volumeLabels) != 0 {
+		if updateVolume {
 			volume.Labels = volumeLabels
 			volume.Spec.RecurringJobs = nil
+
+			log.Infof("Updating %v volume labels to %v", volume.Name, volume.Labels)
```
Comment (shuo-wu) [inline upgrade/v110to120/upgrade.go:275]: NIT: Please add `upgradeLogPrefix`.

**Review by shuo-wu — COMMENTED** [review]
Others LGTM

File: upgrade/v110to120/upgrade.go (hunk 1 above)
Comment (c3y1huang) [reply upgrade/v110to120/upgrade.go]: Addressed.

**Review by shuo-wu — APPROVED** [review]

**Review by innobead — APPROVED** [review]

**Comment by innobead** [comment]
@c3y1huang Please backport to 1.2.x.

**Pull request merged**